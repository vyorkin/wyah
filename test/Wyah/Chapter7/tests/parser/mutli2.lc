let f = (\x -> if x then x else (x+1));
let g = (\x -> if x then 1 else (x+1));
let h = (\x -> if x then (x+1) else 1);
let j = (\x -> fix (x 0));
