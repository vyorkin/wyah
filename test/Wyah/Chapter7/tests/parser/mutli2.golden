Program [Decl "j" (ELam (Var {unVar = "x"}) (EFix (EApp (EVar (Var {unVar = "x"})) (ELit (LInt 0))))),Decl "h" (ELam (Var {unVar = "x"}) (EIf (EVar (Var {unVar = "x"})) (EOp Add (EVar (Var {unVar = "x"})) (ELit (LInt 1))) (ELit (LInt 1)))),Decl "g" (ELam (Var {unVar = "x"}) (EIf (EVar (Var {unVar = "x"})) (ELit (LInt 1)) (EOp Add (EVar (Var {unVar = "x"})) (ELit (LInt 1))))),Decl "f" (ELam (Var {unVar = "x"}) (EIf (EVar (Var {unVar = "x"})) (EVar (Var {unVar = "x"})) (EOp Add (EVar (Var {unVar = "x"})) (ELit (LInt 1)))))]